packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -184,7 +184,9 @@ module.exports = {
     // TODO: consider separate config for production,
     // e.g. to enable no-console and no-debugger only in production.
     configFile: path.join(__dirname, '../.eslintrc'),
-    useEslintrc: false
+    useEslintrc: false,
+    failOnWarning: !!process.env.CI,
```

Comment:
We need to make sure this works after ejecting too.
So `// @remove-on-eject-begin` needs to move inside `eslint` config object.